added documentation for Database util (#3661)
larshp/abapGit: docs/ref-database-util.md

---
title: Database util
category: reference
order: 99
---

> The so-called `Database util` is a tool for managing database entries created by abapGit. You can access the tool via the tools icon in the top right corner of the home page of the abapGit report > choose first menu item `Database util`.

Reviewer: ```suggestion
`Database util` is a tool for managing database entries created by abapGit. You can access the tool via the tools icon in the top right corner of the abapGit home page > choose first menu item `Database util`.
```

It's possible to edit database entries of type

- `USER` and
- `REPO`.

> An `USER` entry does contain meta information like the favorites of an user and its repository configuration e. g. name and email address for Git. This is an example of an `USER` entry:

Reviewer 2: Small spelling/grammar suggestions (not a native speaker so others may correct me as well):
```suggestion
A `USER` entry contains meta information like the favorites of a user and their repository configurations e. g. name and email address for Git. This is an example of a `USER` entry:
```

       <FILENAME>z_my_report.prog.abap</FILENAME>
       <SHA1>[...]</SHA1>
      </item>
      <item>
       <PATH>/src/abaplinted_sample/</PATH>
       <FILENAME>z_my_report.prog.xml</FILENAME>
       <SHA1>[...]</SHA1>
      </item>
     </FILES>
    </item>
   </LOCAL_CHECKSUMS>
   <DOT_ABAPGIT>
    <MASTER_LANGUAGE>E</MASTER_LANGUAGE>
    <STARTING_FOLDER>/src/</STARTING_FOLDER>
    <FOLDER_LOGIC>PREFIX</FOLDER_LOGIC>
    <IGNORE>
     <item>/.gitignore</item>
     <item>/LICENSE</item>
     <item>/README.md</item>
     <item>/package.json</item>
     <item>/.travis.yml</item>
     <item>/.gitlab-ci.yml</item>
     <item>/abaplint.json</item>
     <item>/azure-pipelines.yml</item>
     <item>/src/.gitkeep</item>
    </IGNORE>
    <REQUIREMENTS/>
   </DOT_ABAPGIT>
   <HEAD_BRANCH/>
   <LOCAL_SETTINGS>
    <DISPLAY_NAME>My Test Repository</DISPLAY_NAME>
    <IGNORE_SUBPACKAGES/>
    <WRITE_PROTECTED/>
    <ONLY_LOCAL_OBJECTS/>
    <CODE_INSPECTOR_CHECK_VARIANT>ZMY_CHECK_VARIANT</CODE_INSPECTOR_CHECK_VARIANT>
    <BLOCK_COMMIT>X</BLOCK_COMMIT>
    <SERIALIZE_MASTER_LANG_ONLY/>
   </LOCAL_SETTINGS>
  </REPO>
 </asx:values>
</asx:abap>
```

> If you remove a repository entry from the database util, the repository is not shown in abapGit anymore. This is useful if you want to remove a repository which has a non-existing package assigned and thus can't be clicked on to be select it and delete it. This is like a workaround for such a case.

Reviewer: ```suggestion
If you remove a repository entry from the database util, the repository is not shown in abapGit anymore. This is useful for example if you want to remove a repository which has a non-existing package assigned and thus can't be opened and deleted normally.
```
